torch_xla/csrc/ops/index_ops.cpp

```diff
@@ -96,25 +96,30 @@ IndexAdjacencyInfo GetIndexAdjacencyInfo(at::TensorList indices) {
 //
 // This is a simplified version of at::native::transposeToFront which better
 // fits our requirements.
-CanonicalIndexInfo TransposeToFront(at::Tensor base, at::TensorList indices) {
+CanonicalIndexInfo TransposeToFront(at::Tensor base,
+                                    at::ITensorListRef indices) {
   std::vector<int64_t> dims;
   std::vector<at::Tensor> transposed_indices;
   size_t base_rank = base.dim();
   dims.reserve(base_rank);
   XLA_CHECK_LE(indices.size(), base_rank);
-  for (size_t i = 0; i < indices.size(); i++) {
-    if (indices[i].defined()) {
+  size_t i = 0;
+  for (const auto& index : indices) {
+    if (index.defined()) {
       dims.push_back(i);
-      transposed_indices.emplace_back(indices[i]);
+      transposed_indices.emplace_back(index);
     }
+    ++i;
   }
-  for (size_t i = 0; i < indices.size(); i++) {
-    if (!indices[i].defined()) {
+  i = 0;
+  for (const auto& index : indices) {
```

Comment:
does `indices` does not have a `size`? I am wondering why we switch from `i =; i<; i++` to the new syntax of `for(i : is)`

Reply:
It's because the `ITensorListRef` doesn't allow for random access (aka `indices[I]`, which doesn't seem to be needed here). You can just loop over it using iterators (and then it looks like we need the actual offsets too, so I added a counter for that).